### PR TITLE
[FREELDR] Restructure and separate PC and Xbox platform-specific code

### PR DESCRIPTION
**Pull request by binarymaster**
This PR makes FreeLoader smaller by abstracting and separating platform-specific code with `SARCH` cmake variable / definition.

## TODO

Before merging the PR, other PRs listed below should be handled and merged first, because they touch the files I moved in this PR. I will handle commit conflicts here consequently.

- [x] #1777
- [x] #1782
- [ ] #1958
- [x] #1971
- [x] #2097
- [x] #2167

### CODE REVIEW

File: boot/freeldr/freeldr/arch/i386/xbox/machxbox.c [hunk 1]
```diff
@@ -253,8 +253,22 @@ VOID XboxHwIdle(VOID)
...
+    memset(&MachVtbl, 0, sizeof(MACHVTBL));
+
+    /* Check for Xbox by identifying device at PCI 0:0:0, if it's
+     * 0x10DE/0x02A5 then we're running on an Xbox */
+    WRITE_PORT_ULONG((ULONG *)0xCF8, CONFIG_CMD(0, 0, 0));
+    PciId = READ_PORT_ULONG((ULONG *)0xCFC);
+    if (PciId != 0x02A510DE)
+    {
+        ERR("This is not original Xbox!\n");
+        return;
```
Comment (disean) [inline boot/freeldr/freeldr/arch/i386/xbox/machxbox.c]: `_disable();`
`__halt();`
`for (;;);`

Comment (binarymaster) [reply boot/freeldr/freeldr/arch/i386/xbox/machxbox.c]: Improved

**Comment by binarymaster** [comment]
@Extravert-ir Travis CI is mad at me... it checks formatting of files I just moved but not changed.

**Comment by Mouvedia** [comment]
Does this PR resolve CORE-15427?

**Comment by binarymaster** [comment]
> Does this PR resolve CORE-15427?

Likely yes.

**Comment by binarymaster** [comment]
> Does this PR resolve CORE-15427?

Likely yes, but it's a workaround.

**Comment by disean** [comment]
I would suggest to add `hwide.c` only into Xbox ARC source.

**Comment by HBelusca** [comment]
@disean : it could also be used in PC architecture so its current placement should be fine.

**Pull request merged**